### PR TITLE
feat: bingads audience adding support for multiple action polling and response parsing

### PR DESCRIPTION
**Pull request by shrouti1507**
# Description

< Replace with adequate description for this PR as per [Pull Request document](https://www.notion.so/rudderstacks/Pull-Requests-40a4c6bd7a5e4387ba9029bab297c9e3) >

## Notion Ticket

< Replace with Notion Link >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go [hunk 1]
```diff
@@ -55,94 +55,152 @@ func generateClientID(user User, metadata Metadata) string {
 	return strconv.FormatInt(jobId, 10) + "<<>>" + user.HashedEmail
 }
 
-func (b *BingAdsBulkUploader) CreateZipFile(filePath, audienceId string) (string, []int64, []int64, error) {
-	failedJobIds := []int64{}
-	successJobIds := []int64{}
+func csvZipFileCreator(audienceId string, fileType string) (string, *csv.Writer, string) {
```
Comment (koladilip) [inline router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go:58]: fileType is same as action type? then please use that as name

File: router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go [hunk 2]
```diff
@@ -55,94 +55,152 @@ func generateClientID(user User, metadata Metadata) string {
...
 
-	return zipFilePath, successJobIds, failedJobIds, nil
+	if err != nil {
+		return [3]string{"", "", ""}, successJobIds, failedJobIds, err
+	}
+	var csvWriter [3]*csv.Writer
+	var csvFilePaths [3]string
+	var zipFilePaths [3]string
+	var eventCount [3]int
+	csvFilePaths[0], csvWriter[0], zipFilePaths[0] = csvZipFileCreator(audienceId, "Add")
```
Comment (koladilip) [inline router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go:132]: it is not necessary that always we have all of them everytime

File: router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go [hunk 3]
```diff
@@ -55,94 +55,152 @@ func generateClientID(user User, metadata Metadata) string {
...
+		}
+
+		payloadSizeStat := stats.Default.NewTaggedStat("payload_size", stats.HistogramType, map[string]string{
+			"module":   "batch_router",
+			"destType": b.destName,
+		})
+		payloadSizeStat.Observe(float64(len(data.Message.List)))
+		switch data.Message.Action {
+		case "Add":
+			fileSize[0] = fileSize[0] + len([]byte(line))
```
Comment (koladilip) [inline router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go:152]: this code should move to function, duplicated 3 times

File: router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go (hunk 2 above)
Comment (koladilip) [inline router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go:132]: we should do this using for loop

**Review by koladilip — COMMENTED** [review]
will take look at more deeply

File: router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go (hunk 1 above)
Comment (shrouti1507) [reply router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go:58]: @ItsSudip

File: router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go (hunk 2 above)
Comment (shrouti1507) [reply router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go:132]: @ItsSudip

File: router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go (hunk 3 above)
Comment (shrouti1507) [reply router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go:152]: @ItsSudip

File: router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go (hunk 2 above)
Comment (shrouti1507) [reply router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go:132]: @ItsSudip

Comment (ItsSudip) [reply router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go:132]: I am creating a placeholder for them only. I already removed the file whenever not required.

**Pull request merged**